packages/flutter_tools/lib/src/macos/xcdevice.dart

```diff
@@ -22,11 +22,47 @@ import '../ios/mac.dart';
 import '../reporting/reporting.dart';
 import 'xcode.dart';
 
-enum XCDeviceEvent {
+class XCDeviceEventNotification {
+  XCDeviceEventNotification(
+    this.eventType,
+    this.eventInterface,
+    this.deviceIdentifier,
+  );
+
+  final XCDeviceEventType eventType;
+  final XCDeviceEventInterface eventInterface;
+  final String deviceIdentifier;
+}
+
+enum XCDeviceEventType {
   attach,
   detach,
 }
 
+enum XCDeviceEventInterface {
+  usb,
+  wifi,
+}
+
+String getXCDeviceEventInterface(XCDeviceEventInterface eventInterface) {
```

Comment:
Check out enhanced enums to hang strings and other enums off of specific enum values instead of this pattern: https://dart.dev/guides/language/language-tour#enumerated-types